sympy/assumptions/satask.py

```diff
@@ -332,7 +334,12 @@ def get_all_relevant_facts(proposition, assumptions, context,
 
     if use_known_facts:
         known_facts_CNF = CNF()
-        known_facts_CNF.add_clauses(get_all_known_facts())
+        for expr in all_exprs:
+            if expr.kind == MatrixKind(NumberKind):
+                known_facts_CNF.add_clauses(get_all_known_matrix_facts())
+            elif expr.kind == NumberKind:
+                known_facts_CNF.add_clauses(get_all_known_number_facts())
```

Comment:
This logic does not look write to me. If this does what I think it does, then the clauses are going to be WAY bigger and this is going to be a lot slower.

I recommend using a debugger to go through the code so you can better understand how it works. The part of the code you're modifying here shouldn't be changed. It's the following chunk of code that needs modification:
https://github.com/sympy/sympy/blob/a76b02fcd3a8b7f79b3a88df50c19eb7aee33c17/sympy/assumptions/satask.py#L347-L355